sql: improve SHOW TABLES to show schema and type

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/46553. 

Release justification: none - aiming for 20.2 release.

Release note (sql change): This change modifies SHOW TABLES to return
schema and table type. Furthermore, sequences will now appear in SHOW
TABLES. Any user who relies on SHOW TABLES to return one column can use
`SELECT table_name FROM [SHOW TABLES]` to get compatible behavior
between versions.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46598)
<!-- Reviewable:end -->

otan committed:
sql: improve SHOW TABLES to show schema and type

Release justification: none - aiming for 20.2

Release note (sql change): This change modifies SHOW TABLES to return
schema and table type. Furthermore, sequences will now appear in SHOW
TABLES. Any user who relies on SHOW TABLES to return one column can use
`SELECT table_name FROM [SHOW TABLES]` to get compatible behavior
between versions.
